### PR TITLE
Upgrade to Kubernetes 1.25

### PR DESCRIPTION
**Pull request by shyamradhakrishnan**
**What this PR does / why we need it**:

Upgrade to Kubernetes 1.25

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Review by joekr — APPROVED** [review]

File: test/e2e/data/ccm/ccm.yaml [hunk 1]
```diff
@@ -132,6 +132,8 @@ rules:
       - "extension-apiserver-authentication"
     verbs:
       - get
+      - list
+      - watch
```
[the remark this answers is not in the document]
Comment (shyamradhakrishnan) [reply test/e2e/data/ccm/ccm.yaml:136]: it is there in the diff right? can you please check? - https://github.com/oracle/cluster-api-provider-oci/pull/220/files#diff-8b6d8ebc20d71193e33cff6f83dd790e2c572af22b8f6f4b03a19543a6ea80beR374 or did you mean a different diff?

File: test/e2e/managed_cluster_test.go [hunk 2]
```diff
@@ -254,7 +254,7 @@ func upgradeControlPlaneVersionSpec(ctx context.Context, lister client.Client, c
 	Log(fmt.Sprintf("Upgrade test is starting, upgrade version is %s", managedKubernetesUpgradeVersion))
 	controlPlane.Spec.Version = &managedKubernetesUpgradeVersion
 	Expect(patchHelper.Patch(ctx, controlPlane)).To(Succeed())
-	Log("Upgrade test is starting")
+	Logf("Upgrade test is starting")
```
[the remark this answers is not in the document]
Comment (shyamradhakrishnan) [reply test/e2e/managed_cluster_test.go]: removed

**Comment by shyamradhakrishnan** [comment]
e2e
```
[SynchronizedAfterSuite] PASSED [0.905 seconds]
[SynchronizedAfterSuite]
/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/test/e2e/e2e_suite_test.go:252

  Timeline >>
  STEP: Tearing down the management cluster @ 03/02/23 05:24:42.262
  << Timeline
------------------------------

Ran 6 of 25 Specs in 2853.370 seconds
SUCCESS! -- 6 Passed | 0 Failed | 0 Pending | 19 Skipped
```

File: test/e2e/data/ccm/ccm.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (joekr) [reply test/e2e/data/ccm/ccm.yaml:136]: a probably wrong diff

**Review by joekr — APPROVED** [review]

**Pull request merged**